### ISSUE TITLE
Missing type definition for bind

### ISSUE THREAD

**Issue by royston-c** [flekschas/svelte-simple-modal#58]
The typescript definition is missing for the `bind` function that can be imported using `import { bind } from 'svelte-simple-modal';`

This worked in: `1.0.3`
Throws error in: `1.1.1`

Error: 
```
Module '"svelte-simple-modal"' has no exported member 'bind'. Did you mean to use 'import bind from "svelte-simple-modal"' instead?
```

I can't say exactly the version that is started to fail with but it did work with `1.0.3`.

Workaround: downgrade to 1.0.3

**Comment by flekschas** [comment]
Type definitions were only added in v1.1.1. Before that they simply did not exist.

https://github.com/flekschas/svelte-simple-modal/commit/1b5bcd4184770108a28500bbee1b81ec0a3f8d52
https://unpkg.com/browse/svelte-simple-modal@1.1.1/types/Modal.svelte.d.ts

Are other type annotations found?

**Comment by flekschas** [comment]
If you have the time, it'd be great if you could put together a simple test demo or share your setup for a reproducible example.

**Comment by royston-c** [comment]
```ts
import type { SvelteComponent } from 'svelte';
import { bind } from 'svelte-simple-modal'; // <-- This will error

export const modals = writable<{ resolver: Function, modal: any }[]>([]);

export function showModal<TResponse>(component: typeof SvelteComponent, props?: Record<string, any>): Promise<TResponse | undefined> {
    var resolver;
    var promise = new Promise<TResponse>((resolve) => resolver = resolve);
    modals.update((current) => {
        current.push({ modal: bind(component, props), resolver: resolver });
        return current;
    });

    return promise;
}
```

The above is a small example of the issue with v1.1.1.

Also on a side note, I was wondering if your `Model.svelte.d.ts` should be using `SvelteComponent` (from `import type { SvelteComponent } from 'svelte';`) instead of the untyped `Component` type?

**Comment by flekschas** [comment]
Thanks for the example. You might be right. However, the `Modal.svelte.d.ts` file is auto-generated with [sveld](https://github.com/carbon-design-system/sveld).

**Comment by flekschas** [comment]
I've [opened a ticket with Sveld](https://github.com/carbon-design-system/sveld/issues/56) to see if there's a way the type annotations can be generated properly without requiring manually creating the types.

Apart from that `bind()` is now properly exported in v1.1.2

**Comment by flekschas** [comment]
@royston-c The type def issue should finally be fixed in v1.3.1 thanks to the guys who fixed `sveld`. If you confirm I'll close the issue (or you can close it if you like)

**Comment by flekschas** [comment]
Closing this as I believe the issue is addressed. If it's not, please reopen. Thanks for reporting the problem!
